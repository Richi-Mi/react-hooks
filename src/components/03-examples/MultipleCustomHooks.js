import React from 'react'
import useCounter from '../../hooks/useCounter'
import useFetch from '../../hooks/useFetch'

import '../02-useEffect/effects.css'

const MultipleCustomHooks = () => {
    const { counter, increment } = useCounter(1);
    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`);
    const { author, quote } = !!data && data[0];

    // null == null
    // !null == true
    // !!null == false  
    
    return (
        <div className='multiple_component'>
            <h1> Breakin Bad Quotes </h1>
            <hr />

            {
                loading === true ? (
                    <div className='alert alert-info text-center'>
                        Loading...
                    </div>
                ) : (
                    <blockquote className='blockquote text-right'>
                        <p className='mb-3'> { quote } </p>
                        <footer className='blockquote-footer'> { author } </footer>
                    </blockquote>
                )
            }
            <button className='btn btn-primary' onClick={ () => increment(1)}> Siguiente Quote </button>
            
        </div>
    )
}

export default MultipleCustomHooks