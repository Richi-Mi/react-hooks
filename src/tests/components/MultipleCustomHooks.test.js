import React from 'react';
import { shallow } from "enzyme";
import MultipleCustomHooks from "../../components/03-examples/MultipleCustomHooks";
import useFetch from '../../hooks/useFetch';
import useCounter from '../../hooks/useCounter';

jest.mock('../../hooks/useFetch');
jest.mock('../../hooks/useCounter');

describe(' * Pruebas en el componente <MultipleCustomHooks /> ', () => {

    beforeEach( () => {
        useCounter.mockReturnValue({
            counter: 10
        });
    })

    test(' - Debe de mostrar el componente correctamente', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHooks /> );
        expect( wrapper ).toMatchSnapshot();
        
    });
    test(' - Debe de mostrar la informaciÃ³n cuando ya se tiene algo', () => {
        useFetch.mockReturnValue({
            data: [{
                author: 'Alguien',
                quote: 'Hola Mundo'
            }],
            loading: false,
            error: null
        });
        const wrapper = shallow( <MultipleCustomHooks /> );
        // console.log( wrapper.html() )
        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('.mb-3').text().trim() ).toBe('Hola Mundo');
        expect( wrapper.find('footer').text().trim() ).toBe('Alguien');

    })
    
});
