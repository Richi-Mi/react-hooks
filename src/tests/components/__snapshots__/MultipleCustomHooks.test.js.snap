// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` * Pruebas en el componente <MultipleCustomHooks />   - Debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
     Breakin Bad Quotes 
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
     Siguiente Quote 
  </button>
</div>
`;
